sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/intervalExpressions.scala

```diff
@@ -110,50 +110,57 @@ object ExtractIntervalPart {
   }
 }
 
-abstract class IntervalNumOperation(
```

Comment:
We can still implement everything in `IntervalNumOperation`, the subclasses only need to provide the method.